### PR TITLE
fix open files issue and bump firesim with various fixes

### PR DESCRIPTION
**Pull request by sagark**
Fix the open files limit issue automatically for systems that have a sufficient hard limit, but insufficient soft limit (e.g. all millennium machines).

Also, many fixes from https://github.com/firesim/firesim/pull/1266

**Related PRs / Issues**:
N/A

**Type of change**:
- [x] Bug fix
- [ ] New feature
- [ ] Other enhancement

**Impact**:
- [ ] RTL change
- [ ] Software change (RISC-V software)
- [ ] Build system change
- [x] Other

**Contributor Checklist**:
- [x] Did you set `main` as the base branch?
- [x] Is this PR's title suitable for inclusion in the changelog and have you added a `changelog:<topic>` label?
- [x] Did you state the type-of-change/impact?
- [x] Did you mark the PR with a `changelog:` label?

### CODE REVIEW

**Review by abejgonzalez — DISMISSED** [review]

File: scripts/fix-open-files.sh [hunk 1]
```diff
@@ -0,0 +1,12 @@
+
```
[the remark this answers is not in the document]
Comment (sagark) [reply scripts/fix-open-files.sh:1]: This file only makes sense when sourced.

File: scripts/fix-open-files.sh [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+# first, check if the system allows sufficient limits (the hard limit)
+HARD_LIMIT=$(ulimit -Hn)
+REQUIRED_LIMIT=16384
+
+if [ "$HARD_LIMIT" -lt "$REQUIRED_LIMIT" ]; then
+    echo "WARNING: Your system does not support an open files limit (the output of 'ulimit -Sn' and 'ulimit -Hn') of at least $REQUIRED_LIMIT, which is required to workaround a bug in buildroot. You will not be able to build a Linux distro with FireMarshal until this is addressed."
+fi
+
+# in any case, set the soft limit to the same value as the hard limit
+ulimit -Sn $(ulimit -Hn)
```
[the remark this answers is not in the document]
Comment (sagark) [reply scripts/fix-open-files.sh:11]: No

File: scripts/first-clone-setup-fast.sh [hunk 3]
```diff
@@ -0,0 +1,33 @@
+#!/usr/bin/env bash
+
+set -ex
+
```
Comment (jerryz123) [inline scripts/first-clone-setup-fast.sh:4]: Do we have to have **Y**et **A**nother **S**etup **S**cript?

Comment (jerryz123) [reply scripts/first-clone-setup-fast.sh:4]: Can't the parts of the script be wrapped into build-setup?

Comment (jerryz123) [reply scripts/first-clone-setup-fast.sh:4]: Perhaps the right thing to do is to add a `--firesim` flag to `build-setup.sh` that also inits firesim and firemarshal.

Comment (jerryz123) [reply scripts/first-clone-setup-fast.sh:4]: The plan is to merge this script now as a temporary thing, then by the next release, the top level setup script will be modified to also do everything this script currently does.

**Comment by jerryz123** [comment]
The plan is to merge this script now as a temporary thing, then by the next release, the top level setup script will be modified to also do everything this script currently does.

**Review by jerryz123 — APPROVED** [review]
Approved, with the understanding that the first-setup-fast script will be merged into the build-setup script in future PRs

**Pull request merged**